File: java-quarkus/code-with-quarkus/src/main/java/org/acme/model/Rectangle.java
package org.acme.model;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

public class Rectangle implements FigureInterface{
    private final int sizeA;
    private final int sizeB;

    @JsonCreator
    Rectangle(@JsonProperty("sizeA") int sizeA, @JsonProperty("sizeB") int sizeB) {
        this.sizeA = sizeA;
        this.sizeB = sizeB;
    }

    @Override
    public String getType() {
        return "rectangle";
    }

    @Override
    public double computeArea() {
        return sizeA * sizeB;
    }
}

File: kotlin-ktor-jvm/src/main/kotlin/com/example/model/Rectangle.kt
package com.example.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
@SerialName("rectangle")
class Rectangle(
    private val sizeA: Int,
    private val sizeB: Int,
): Figure {
    override fun getType(): String {
        return "rectangle";
    }

    override fun computeArea(): Double {
        return sizeA.toDouble() * sizeB.toDouble();
    }
}